test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -368,14 +423,159 @@ def test_generator(setup_and_teardown):
+        [2, 'assembly_2', 'KBaseGenomeAnnotations.Assembly-10', 'time', 1, 'user', 42, 'wsname', 'md5', 78, {}]
+    ]
+
+    genome_obj_infos = [
+        [3, 'genome_1', 'KBaseGenomes.Genome-6', 'time', 1, 'user', 42, 'wsname', 'md5', 78,
+         {"load_id": load_id, "Assembly Object": "1/1/1"}],
+        [4, 'genome_2', 'KBaseGenomes.Genome-9.3', 'time', 1, 'user', 42, 'wsname', 'md5', 78,
+         {"load_id": load_id, "Assembly Object": "2/2"}]
+    ]
+
+    wait_to_upload_genomes = {
+        genome_name: genome_dir
+        for genome_name, genome_dir in zip(GEMOME_NAMES, genome_collection_source_dirs)
+    }
+
+    # ws.get_object_info3() is unused in this test case
+    ws = create_autospec(Workspace, spec_set=True, instance=True)
+    gfu = create_autospec(GenomeFileUtil, spec_set=True, instance=True)
+    genbanks_to_genomes_results = {
+        "results": [{"genome_ref": "42/3/1",
+                     "assembly_ref": "42/1/1",
+                     "assembly_path": job_dir / ASSEMBLY_NAMES[0],
+                     "assembly_info": assembly_obj_infos[0],
+                     "genome_info": genome_obj_infos[0]},
+                    {"genome_ref": "42/4/1",
+                     "assembly_ref": "42/2/1",
+                     "assembly_path": job_dir / ASSEMBLY_NAMES[1],
+                     "assembly_info": assembly_obj_infos[1],
+                     "genome_info": genome_obj_infos[1]}]
+    }
+    gfu.genbanks_to_genomes.return_value = genbanks_to_genomes_results
+
+    with patch.object(workspace_uploader, '_JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER', new=job_dir):
+        uploaded_count = workspace_uploader._upload_objects_in_parallel(
+            ws=ws,
+            upload_env_key="CI",
+            workspace_id=42,
+            load_id="214",
+            collections_source_dir=collection_source_dir,
+            wait_to_upload_objs=wait_to_upload_genomes,
+            batch_size=2,
+            source_data_dir=sourcedata_dir,
+            asu_client=Mock(),
+            gfu_client=gfu,
+            job_data_dir=job_dir,
+            upload_assembly_only=False,
+        )
+
+    assert uploaded_count == 2
+
+    # assert that no interactions occurred with ws
+    ws.get_object_info3.assert_not_called()
+
+    # assert that asu was called correctly
+    gfu.genbanks_to_genomes.assert_called_once_with(
+        {
+            "workspace_id": 42,
+            "inputs": [
+                {
+                    "file": {"path": f"{job_dir}/GCF_000979855.1_gtlEnvA5udCFS_genomic.gbff.gz"},
+                    "genome_name": "GCF_000979855.1_gtlEnvA5udCFS_genomic.gbff.gz",
+                    "metadata": {"load_id": "214"},
+                },
+                {
+                    "file": {"path": f"{job_dir}/GCF_000979175.1_gtlEnvA5udCFS_genomic.gbff.gz"},
+                    "genome_name": "GCF_000979175.1_gtlEnvA5udCFS_genomic.gbff.gz",
```

Comment:
These should be genome_1 and genome_2 per the objinfo unless I'm misunderstanding

Reply:
those should be the file names. We assume file name and obj names are the same.

Reply:
Ok, so `genome_1` etc should be these then?

Reply:
yea. updated obj_info.

Reply:
:+1: